python/amici/petab_import_pysb.py

```diff
@@ -0,0 +1,333 @@
+                    p = pysb.Parameter(str(s), 1.0)
+                    locals[sp.Symbol.__str__(p)] = p
+
+        # add observables and sigmas to pysb model
+        for (observable_id, observable_formula, noise_formula) \
+                in zip(self.observable_df.index,
+                       self.observable_df[petab.OBSERVABLE_FORMULA],
+                       self.observable_df[petab.NOISE_FORMULA]):
+            # No observableTransformation so far
+            if petab.OBSERVABLE_TRANSFORMATION in self.observable_df:
+                trafo = self.observable_df.loc[observable_id,
+                                               petab.OBSERVABLE_TRANSFORMATION]
+                if trafo and trafo != petab.LIN:
+                    raise NotImplementedError(
+                        "Observable transformation currently unsupported "
+                        "for PySB models")
+            obs_symbol = sp.sympify(observable_formula, locals=locals)
+            obs_expr = pysb.Expression(observable_id, obs_symbol)
+            locals[observable_id] = obs_expr
+
+            sigma_id = f"{observable_id}_sigma"
+            sigma_symbol = sp.sympify(
+                noise_formula,
+                locals=locals
+            )
+            sigma_expr = pysb.Expression(sigma_id, sigma_symbol)
+            locals[sigma_id] = sigma_expr
+
+    @staticmethod
+    def from_files(sbml_file: str = None,
+                   condition_file: str = None,
+                   measurement_file: Union[str, Iterable[str]] = None,
+                   parameter_file: Union[str, List[str]] = None,
+                   visualization_files: Union[str, Iterable[str]] = None,
+                   observable_files: Union[str, Iterable[str]] = None,
+                   pysb_model_file: str = None,
+                   ) -> 'PysbPetabProblem':
+        """
+        Factory method to load model and tables from files.
+
+        Arguments:
+            sbml_file: PEtab SBML model
+            condition_file: PEtab condition table
+            measurement_file: PEtab measurement table
+            parameter_file: PEtab parameter table
+            visualization_files: PEtab visualization tables
+            observable_files: PEtab observables tables
+            pysb_model_file: PySB model file
+        """
+
+        sbml_model = sbml_document = sbml_reader = None
+        condition_df = measurement_df = parameter_df = visualization_df = None
+        observable_df = None
+
+        if condition_file:
+            condition_df = petab.conditions.get_condition_df(condition_file)
+
+        if measurement_file:
+            # If there are multiple tables, we will merge them
+            measurement_df = petab.core.concat_tables(
+                measurement_file, petab.measurements.get_measurement_df)
+
+        if parameter_file:
+            parameter_df = petab.parameters.get_parameter_df(parameter_file)
+
+        if sbml_file:
```

Comment:
I thought the SBML dummy is created in `PysbPetabProblem`?

Reply:
Right. Cleanup up a bit.